equinix/resource_metal_device_acc_test.go

```diff
@@ -118,6 +118,14 @@ resource "random_integer" "plan_idx" {
   max = length(data.equinix_metal_plans.test.plans) - 1
 }
 
+resource "terraform_data" "plan" {
+  input = data.equinix_metal_plans.test.plans[random_integer.plan_idx.result]
+
+  lifecycle {
+	ignore_changes = ["input"]
```

Comment:
This is a nice trick to effectively cache the result of the data source. Any good links to check out (curious if there are other tricks like this given).

Reply:
The only reference I found for this resource was [the docs page on the terraform developer site](https://developer.hashicorp.com/terraform/language/resources/terraform-data).  It doesn't specifically cover this use case of caching values, but it does include a similar trick for using `lifecycle` to trigger replacement of a different resource.